dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/monitor/DatadogRumMonitor.kt

```diff
@@ -14,15 +14,20 @@ import com.datadog.android.rum.RumMonitor
 import com.datadog.android.rum.RumResourceKind
 import com.datadog.android.rum.internal.domain.RumEvent
 import com.datadog.android.rum.internal.domain.RumEventData
+import com.datadog.android.rum.internal.domain.RumEventSerializer
+import java.lang.ref.WeakReference
 import java.util.UUID
 
 internal class DatadogRumMonitor(
     private val timeProvider: TimeProvider,
     private val writer: Writer<RumEvent>
 ) : RumMonitor {
 
-    private val activeView = mutableMapOf<Any, RumEvent>()
-    private val activeResources = mutableMapOf<Any, RumEvent>()
+    private var activeViewEvent: RumEvent? = null
+    private var activeViewData: RumEventData.View? = null
+    private var activeViewKey: WeakReference<Any?> = WeakReference(null)
+
+    private val activeResources = mutableMapOf<WeakReference<Any>, RumEvent>()
```

Comment:
instead of this you could use a WeakHashMap  same for the activeViewEvent.

Reply:
A weak hash map hides a lot in its implementation (and Android's implementation differs from the JDK, meaning the tests I did weren't consistent). Among other things, I want to be able to actually manipulate the WeakReference, which you can't do if you use a WeakHashMap.